filesystem/trd.ksy

```diff
@@ -0,0 +1,138 @@
+  floppies it's just a physical track number, for two-side floppies sides are
+  interleaved, i.e. logical_track_num = (physical_track_num << 1) | side
+
+  So, this format definition is more for TR-DOS filesystem than for .trd files,
+  which are formatless.
+
+  Strings (file names, disk label, disk password) are padded with spaces and use
+  ZX Spectrum character set, including UDGs, block drawing chars and Basic
+  tokens. ASCII range is mostly standard ASCII, with few characters (^, `, DEL)
+  replaced with (up arrow, pound, copyright symbol).
+seq:
+  - id: files
+    type: file
+    repeat: until
+    # After 128 files there is disk info entry, which also has 0x00 terminator
+    # in the same position as file name. So usually even with 128 files you can
+    # just read until 0x00.
+    repeat-until: _.is_terminator
+instances:
+  volume_info:
+    pos: 0x800
+    type: volume_info
+types:
+  file:
+    seq:
+      - id: name
+        type: str
+        size: 8
+      - id: extension
+        type: str
+        size: 1
+
+      - id: program_and_data_length
+        type: u2
+        if: extension == "B"
+      - id: start_address
+        doc: Default memory address to load this byte array into
+        type: u2
+        if: extension == "C"
+      - id: print_extent_num
+        type: u1
+        if: extension == "#"
+      - id: print_unused
+        type: u1
+        if: extension == "#"
+      - id: length1_unused
+        size: 2
+        if: extension != "B" and extension != "C" and extension != "#"
+
+      - id: length
+        type: u2
+        doc: |
+          For "B" type, it's length of program only, use
+          program_and_data_length for full length. For other types it's length
+          in bytes
+      - id: length_sectors
+        type: u1
+      - id: starting_sector
+        type: u1
+      - id: starting_track
+        type: u1
+    instances:
+      is_deleted:
+        value: name.substring(0, 1) == "\1"
+      is_terminator:
+        value: name.substring(0, 1) == "\0"
+      contents:
+        pos: starting_track * 256 * 16 + starting_sector * 256
+        size: length_sectors * 256
+  volume_info:
+    seq:
+      # This is 0x00 at the same position as first character of filename in
+      # file entries for convenience. When disk has 128 files, it acts as
+      # "last file" terminator.
+      - id: catalog_end
+        contents: [0]
+      - id: unused
+        size: 224
+      - id: first_free_sector_sector
+        type: u1
+      - id: first_free_sector_track
+        doc: |
+          track number is logical, for double-sided disks it's
+          (physical_track << 1) | side, the same way that tracks are stored
+          sequentially in .trd file
+        type: u1
+      - id: disk_type
+        type: u1
+        enum: disk_type
+      - id: number_of_files
+        doc: |
+          Number of non-deleted files. Directory can have more than
+          number_of_files entries due to deleted files
```

Comment:
Please consider following [style guide](http://doc.kaitai.io/ksy_style_guide.html) and naming these `num_files`, `num_free_sectors`, etc?

Reply:
Changed.